ublox/src/ubx_packets/packets.rs

```diff
@@ -1970,15 +1973,55 @@ struct RxmRtcm {
 }
 
 #[ubx_packet_recv]
-#[ubx(class = 0x10, id = 0x02, fixed_payload_len = 16)]
+#[ubx(class = 0x10, id = 0x02, max_payload_len = 1240)]
 struct EsfMeas {
     time_tag: u32,
+    #[ubx(map_type = EsfMeasInfo, from = EsfMeasInfo::new)]
+    info: [u8; 0],
+}
+
+pub struct EsfMeasInfo<'a> {
     flags: u16,
     id: u16,
-    /// flags >> 11 & 0x1F
-    data: u32,
-    /// flags & 0x8
-    calib_tag: u32,
+    data: Box<dyn Iterator<Item = &'a u32>>,
+    calib_tag: Option<u32>,
+}
+
+impl<'a> EsfMeasInfo<'a> {
+    fn new(bytes: &'a [u8]) -> Self {
+        let mut chunks = bytes.chunks_exact(2);
+        let flags = u16::from_le_bytes(chunks.next().unwrap().try_into().unwrap());
```

Comment:
If there is a way to retrieve this within the param `bytes: &'a [u8]`, then we can keep `flags` and `id` out of the info struct but noooo it decides to trim the bytes as well

Reply:
like the spec, offset is outlined so think the crate should have passed the entire byte instead of just the remaining bytes but maybe reworking that means all the other msgs would have to change as well